src/GslibIO.jl

```diff
@@ -44,6 +52,25 @@ function load(file::File{format"GSLIB"})
   end
 end
 
+# helper function to parse a legacy GSLIB file
+function parse_legacy(filename::AbstractString)
+  open(filename) do fs
+    header = readline(fs)
+    # the second line contains the number of variables (the first integer)
+    # and it may contain extra info (comments), which will be ignored
+    linesplit = split(readline(fs))
+    nvars = parse(Int, linesplit[1])
```

Comment:
So I understand that this line contains an integer followed by a comment. can we maybe rewrite the code so that we just consume the first word in the line and discard the rest? right now we are consuming the entire line, splitting and discarding. we need to double check if the IO library provides a simple `read(fs)` that reads the next word in the file stream.

Reply:
I was thinking the same. Using the split method may not work in cases such as
```HEADER
2 #HOLA
```
I will try to use read(fs) instead.

Reply:
yes, that would be nice. we just need to read the first "word" and then discard the rest with a `readline(fs)` call

Reply:
Surprisingly, I could not find any existing function to scan an integer from a stream. One option would be to write a small automata to detect the first integer number reading character by character, but it seems that we are complicating too much with something that should be simple. In the meantime, I propose to keep the implementation as it is, until we can find the appropriated best way to read an integer. I don't see any way to avoid reading the complete line (except parsing with `read(stream, Char)`), and to be honest, I don't see any relevant cost/benefit.

Reply:
We can at least change the code to:

```julia
words = split(readline(fs))
nvars = parse(Int, words[1])
```

while a solution with a different function isn't ready. I guess that Base contains something, but we can leave the code as is for now, it doesn't change much.

Notice how I tend to choose short identifiers for variables, in this case `words` is clear and makes the code align well with `nvars` below. Let's try to follow this pattern moving forward 👍🏽